go/common/testing/e2e_coverage.go

```diff
@@ -0,0 +1,25 @@
+// +build e2ecoverage
+
+package testing
+
+import (
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/oasislabs/oasis-core/go/common"
+)
+
+// RunMain runs the specified main function inside unit tests. This is usually used in E2E coverage
+// wrapper "tests" in order to generate coverage for a binary.
+func RunMain(t *testing.T, mainFn func()) {
+	// Trim (test-related) arguments and call the usual main.
+	os.Args = common.TrimArgs(os.Args)
+	mainFn()
+
+	// Suppress coverage report on stdout.
+	f, err := os.OpenFile(os.DevNull, os.O_RDWR, 0)
```

Comment:
https://golang.org/pkg/io/ioutil/#pkg-variables

Reply:
Yeah, but you need a file as `os.Stdout` is a file?

Reply:
Oh derp